src/apps/backend/documentation/documentation-service.ts

```diff
@@ -0,0 +1,36 @@
+import fs from 'fs';
+import path from 'path';
+
+import App from '../app';
+import { ConfigService } from '../modules/config';
+import { Logger } from '../modules/logger';
+
+import expressListRoutes from './internals/express-list-routes';
+
+export default class DocumentationService {
+  public static generateDocumentation(): void {
+    const apiMicroservices = App.getAPIMicroservices();
+    // TODO: update to a valid path & file name, once done with documentation generation through ai
+    const documentationPath = path.join(
+      __dirname,
+      '../../../assets/documentation/index.md',
+    );
+    let documentationContent = '';
+
+    apiMicroservices.forEach((server) => {
+      const routes = expressListRoutes(server.serverInstance.server);
+      const serverInfo = `Server Root: ${server.serverRootFolderPath}\nRoutes:\n${JSON.stringify(routes, null, 2)}\n`;
+      documentationContent += serverInfo;
+    });
+
+    const isDocumentationEnabled = ConfigService.getValue<boolean>(
```

Comment:
Probably add a cheeck for this at the beginning of this method and bail if not enabled.